### ISSUE TITLE
The year is incremented wrongly when parsing  last short week  in December for any year.

### ISSUE THREAD

**Issue by igrakhimov** [date-fns/date-fns#831]
The buggy function is webpack:///./~/wc-lib/~/date-fns/esm/_lib/getUTCWeekYear/index.js

E.g. from date picker one choose 26/12/202**1** after parsing it will return 26/12/202**2**

**Comment by leshakoss** [comment]
@igrakhimov could you please provide the exact code line that returns a wrong result?

**Comment by igrakhimov** [comment]
@leshakoss  at line 54 the wrong year returned; at line 46 the startOfNextYear is set (that is where the problem starts I think) basically when at line 53 input date compared with startOfNextYear  the logic should be going to line 55 hope it help 
![image](https://user-images.githubusercontent.com/8219089/43872038-e274fd6c-9bc2-11e8-8f40-6e4e27427257.png)

**Comment by leshakoss** [comment]
@igrakhimov no, I mean the line of your code. Basically, what are you trying to achieve?

**Comment by igrakhimov** [comment]
@leshakoss  sorry for delay, here we go 
function that is calling the 'baggy' one 
********************************************************
import Vue from 'vue';
**import {format} from 'date-fns/esm';**

Vue.filter('formatDate', (value: any, dateFormat?: string) => {
  return formatDate(value, dateFormat);
});
export function formatDate(value: any, dateFormat?: string) : string  {
  if (value) {
    return format(value, dateFormat ? dateFormat : 'dd/MM/YYYY HH:mm:ss');
  }
  return '';
}; 

Basically if input param is 
     30th Dec 2018 respond will be 30/12/201**8**
but for
     31st Dec 2018 respond will be 31/12/201**9** 

Note1: 31st of Dec 2018 is a Monday hence week is splitted between 2018 and 2019 
Note2: for non splitted week there is no issue

**Comment by leshakoss** [comment]
[Since 8th alpha](https://date-fns.org/v2.0.0-alpha.16/docs/parse), `YYYY` is token for [ISO week-numbering year](https://en.wikipedia.org/wiki/ISO_week_date). Please use `yyyy` for regular year.

**Comment by igrakhimov** [comment]
@leshakoss **did try as you suggested - still the same result.** See below 
**date** (input formatted using lower case year yyyy)
Mon Dec 30 2019 11:00:00 GMT+1100 (Australian Eastern Daylight Time)
**startOfNextYear** (set on line 54 in getUTCWeekYear)
Sun Dec 29 2019 11:00:00 GMT+1100 (Australian Eastern Daylight Time)

Hence when comparing dates above at line 53 we have year incremented by 1 at line 54 which is wrong

**Comment by igrakhimov** [comment]
@leshakoss sorry correction: **startOfNextYear** (set on line **46** in getUTCWeekYear)

**Comment by leshakoss** [comment]
@igrakhimov I still don't really understand what you're trying to do.
The lines of date-fns code that you're posting are relevant only for ISO week-numbering years and not for regular years.

I tried to replicate your bug by doing this:

```
lesha@L ~/W/date-fns (master)> env TZ="Australia/Sydney" ./node_modules/.bin/babel-node
> const format = require('./src/format')
undefined
> format(new Date(2018, 11 /* Dec */, 30), 'dd/MM/yyyy')
'30/12/2018'
> format(new Date(2018, 11 /* Dec */, 31), 'dd/MM/yyyy')
'31/12/2018'
```

As far as I can see, the result is correct. Could you please explain me what exact arguments passed to `format` function give a wrong result?

**Comment by igrakhimov** [comment]
@leshakoss Thanks Lesha for your help. You are absolutely right. If year token is yyyy then all is good.
